lib/ts/test-utils.ts

```diff
@@ -0,0 +1,209 @@
+
+                        globalModifiers.forEach((globalModifier) => {
+                            ["", ...variants].forEach((variant) => {
+                                const variantClasses = makeClass(variant);
+                                const classesVariant = `${baseClass} ${variantClasses}${primaryClasses}${secondaryClasses} ${globalModifier}`;
+                                const testidVariant = buildTestid([
+                                    testidBase,
+                                    variant,
+                                    [
+                                        primaryModifier,
+                                        secondaryModifier,
+                                        globalModifier,
+                                    ]
+                                        .filter(Boolean)
+                                        .join("-"),
+                                ]);
+
+                                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                                testVariations.push({
+                                    testid: testidVariant,
+                                    classes: classesVariant,
+                                    theme,
+                                });
+                            });
+                        });
+                    });
+                });
+
+                // create standalone modifiers test props
+                modifiers?.standalone?.forEach((standaloneModifier) => {
+                    const standaloneClasses = makeClass(standaloneModifier);
+                    const testidVariant = buildTestid([
+                        testidBase,
+                        standaloneModifier,
+                    ]);
+
+                    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                    testVariations.push({
+                        testid: testidVariant,
+                        classes: makeClass(standaloneClasses),
+                        theme,
+                    });
+                });
+            }
+        );
+    });
+
+    // Sorting for readability
+    return testVariations.sort((a, b) => a.testid.localeCompare(b.testid));
+};
+
+export const makeTestElement = ({
+    attributes = {},
+    children = "",
+    tag = "div",
+    testid,
+}: {
+    attributes?: Record<string, string>;
+    children?: string;
+    tag?: string;
+    testid: string;
+}) => {
+    const unsafe = {
+        tag: unsafeStatic(tag),
+        attributes: unsafeStatic(attrObjToString(attributes).toString()),
+        children: unsafeStatic(children),
+    };
+
+    return html`
+        <${unsafe.tag}
+            ${unsafe.attributes}
+            data-testid="${testid}"
+        >
+            ${unsafe.children}
+        </${unsafe.tag}>
+    `;
+};
+
+export const makeTest = ({
+    description,
+    element,
+    testid,
+    theme = [],
+    type,
+}: {
+    description?: string;
```

Comment:
Skimming through the code I did not see the description field use. Premature optimization? These are automatically generated tests maybe we should not allow the user to override our automatically generated descriptions.

Reply:
I have it only on the `.s-btn.is-loading` visual test which _hopefully_ will be replaced by generated tests soon. Someone could make an individual test using `makeTest` and pass a description, but yeah it does seem kinda like a premature optimization. I'll remove it!